diff --git a/system-config-kdump.desktop.in b/system-config-kdump.desktop.in
index 0ce351e..5687387 100644
--- a/system-config-kdump.desktop.in
+++ b/system-config-kdump.desktop.in
@@ -1,7 +1,7 @@
 [Desktop Entry]
 _Name=Kernel crash dumps
 _Comment=Configure kernel crash dumps
-Categories=System;Settings;Application;GTK;
+Categories=GNOME;GTK;System;
 Icon=system-config-kdump
 Exec=/usr/bin/system-config-kdump
 Type=Application
